### ISSUE TITLE
Unable to enable IntelliSense auto-complete

### ISSUE THREAD

**Issue by bitinn** [OmniSharp/omnisharp-vscode#2030]
## Environment data
`dotnet --info` output:

```
.NET Command Line Tools (2.1.4)

Product Information:
 Version:            2.1.4
 Commit SHA-1 hash:  5e8add2190

Runtime Environment:
 OS Name:     Mac OS X
 OS Version:  10.13
 OS Platform: Darwin
 RID:         osx.10.12-x64
 Base Path:   /usr/local/share/dotnet/sdk/2.1.4/

Microsoft .NET Core Shared Framework Host

  Version  : 2.0.5
  Build    : 17373eb129b3b05aa18ece963f8795d65ef8ea54
```

VS Code version:

`Version 1.20.0 (1.20.0)`

C# Extension version:

`1.13.1`

## Steps to reproduce

`^Space` to trigger suggestion

## Expected  behavior

Suggestion based on assembly and classes in project.

## Actual behavior

All I see is word-based suggestion (based on words in the same file)

<img width="289" alt="screen shot 2018-02-10 at 22 50 20" src="https://user-images.githubusercontent.com/1484279/36063214-dbd27746-0eb4-11e8-8e54-8433706fecca.png">

<img width="295" alt="screen shot 2018-02-10 at 22 52 32" src="https://user-images.githubusercontent.com/1484279/36063242-2a18368e-0eb5-11e8-85bb-2efa332b89b8.png">

<img width="312" alt="screen shot 2018-02-10 at 22 52 56" src="https://user-images.githubusercontent.com/1484279/36063246-2e5ce1b8-0eb5-11e8-81e4-b8c2bfb91f64.png">

## Additional Comment

- I have C# extension, Unity extension, Mono, dotnet cli installed as suggested by MS doc.
- I have opened a folder, not seeing any "fire" icon to ask me to setup a project, perhaps because it was opened by Visual Studio Community before.
- I saw VSCode downloading OmniSharp package, and it seems to have succeeded.
- My editor settings should allow IntelliSense to show up:

```
  // Controls if suggestions should automatically show up while typing
  "editor.quickSuggestions": {
    "other": true,
    "comments": false,
    "strings": false
  },

  // Controls the delay in ms after which quick suggestions will show up
  "editor.quickSuggestionsDelay": 10,
```

- OmniSharp related config looks like this, I see no option to re-install OmniSharp, nor do I know where VSCode put its packages.

```
  // Specifies whether the OmniSharp server will be automatically started or not. If false, OmniSharp can be started with the 'Restart OmniSharp' command
  "omnisharp.autoStart": true,

  // Specifies the level of logging output from the OmniSharp server.
  "omnisharp.loggingLevel": "information",

  // The maximum number of projects to be shown in the 'Select Project' dropdown (maximum 250).
  "omnisharp.maxProjectResults": 250,

  // Specifies the full path to the OmniSharp server.
  "omnisharp.path": null,

  // The time Visual Studio Code will wait for the OmniSharp server to start. Time is expressed in seconds.
  "omnisharp.projectLoadTimeout": 60,

  // Specifes whether OmniSharp should use VS Code editor settings for C# code formatting (use of tabs, indentation size).
  "omnisharp.useEditorFormattingSettings": true,

  // Launch OmniSharp with Mono.
  "omnisharp.useMono": false,

  // Pass the --debug flag when launching the OmniSharp server to allow a debugger to be attached.
  "omnisharp.waitForDebugger": false
```

- I want a way to start over if fixing this is too much trouble: basically, clean everything VSCode has downloaded and force it setup again?

**Comment by bitinn** [comment]
The crazy thing is: I have the solution file in the folder, when I open the folder (not a single file) everything seem fine but there is no "fire" button to let me pick project at all.

<img width="201" alt="screen shot 2018-02-11 at 1 42 30" src="https://user-images.githubusercontent.com/1484279/36064895-0a9ab698-0ecd-11e8-9abc-293991427f85.png">

I have gotten _Visual Studio Community_ and _Sublime Text with OmniSharp plugin_ to work with this repo, with working auto-complete for .Net Core and Unity code. It's VSCode I have no idea why it isn't working.

- I see no output in logs from OmniSharp server.
- Nor is there any localhost requests in Inspector.
- `~/.vscode/extensions/ms-vscode.csharp-1.13.1/.omnisharp/` appears to have the right files.

**Comment by rchande** [comment]
@bitinn Can you share your omnisharp log? Output window -> select "omnisharp log". I'm wondering if the fact that you set the "omnisharp.path" option to null is a problem--can you comment it out and see if it helps?

**Comment by bitinn** [comment]
@rchande Like all things in life, when you try to reproduce it, it went away.

I clean installed VSCode (removed preference in ~/Library/Application Support and ~/.vscode/) and this time perform installation steps in exact order:

- dotnet
- Mono
- VSCode
- C# Extension
- Unity debugger extension

And then I tried to open up the project, OmniSharp starts up fine (the settings are still the same as posted above).

```
Starting OmniSharp server at 2/13/2018, 2:34:07 PM
    Target: /Users/df/git/test/test.sln

OmniSharp server started wth Mono
    Path: /Users/df/.vscode/extensions/ms-vscode.csharp-1.13.1/.omnisharp/omnisharp/OmniSharp.exe
    PID: 709

﻿[info]: OmniSharp.Stdio.Host
        Starting OmniSharp on MacOS 10.13.3 (x64)
[info]: OmniSharp.MSBuild.Discovery.MSBuildLocator
        Located 2 MSBuild instance(s)
            1: Mono 15.0 - "/Library/Frameworks/Mono.framework/Versions/5.8.0/lib/mono/msbuild/15.0/bin"
            2: StandAlone 15.0 - "/Users/df/.vscode/extensions/ms-vscode.csharp-1.13.1/.omnisharp/omnisharp/msbuild/15.0/Bin"
[info]: OmniSharp.MSBuild.Discovery.MSBuildLocator
        Registered MSBuild instance: Mono 15.0 - "/Library/Frameworks/Mono.framework/Versions/5.8.0/lib/mono/msbuild/15.0/bin"
            CscToolPath = /Users/df/.vscode/extensions/ms-vscode.csharp-1.13.1/.omnisharp/omnisharp/msbuild/15.0/Bin/Roslyn
            CscToolExe = csc.exe
[info]: OmniSharp.Cake.CakeProjectSystem
        Detecting Cake files in '/Users/df/git/test'.
[info]: OmniSharp.Cake.CakeProjectSystem
        Could not find any Cake files
[info]: OmniSharp.DotNet.DotNetProjectSystem
        Initializing in /Users/df/git/test
[info]: OmniSharp.DotNet.DotNetProjectSystem
        Auto package restore: False
[info]: OmniSharp.DotNet.DotNetProjectSystem
        Update workspace context
[info]: OmniSharp.DotNet.DotNetProjectSystem
        Resolving projects references
[info]: OmniSharp.MSBuild.MSBuildProjectSystem
        Detecting projects in '/Users/df/git/test/test.sln'.
[info]: OmniSharp.MSBuild.MSBuildProjectSystem
        Loading project: /Users/df/git/test/Assembly-CSharp-firstpass.csproj
[info]: OmniSharp.MSBuild.MSBuildProjectSystem
        Loading project: /Users/df/git/test/Assembly-CSharp.csproj
[info]: OmniSharp.MSBuild.MSBuildProjectSystem
        Loading project: /Users/df/git/test/Assembly-CSharp-Editor-firstpass.csproj
[info]: OmniSharp.MSBuild.MSBuildProjectSystem
        Loading project: /Users/df/git/test/Assembly-CSharp-Editor.csproj
[info]: OmniSharp.MSBuild.MSBuildProjectSystem
        Update project: Assembly-CSharp-firstpass
[info]: OmniSharp.MSBuild.MSBuildProjectSystem
        Update project: Assembly-CSharp
[warn]: OmniSharp.MSBuild.MSBuildProjectSystem
        Unable to resolve assembly '/Users/df/git/test/Temp/bin/Debug/Assembly-CSharp-firstpass.dll'
[info]: OmniSharp.MSBuild.MSBuildProjectSystem
        Update project: Assembly-CSharp-Editor-firstpass
[warn]: OmniSharp.MSBuild.MSBuildProjectSystem
        Unable to resolve assembly '/Users/df/git/test/Temp/bin/Debug/Assembly-CSharp-firstpass.dll'
[info]: OmniSharp.MSBuild.MSBuildProjectSystem
        Update project: Assembly-CSharp-Editor
[warn]: OmniSharp.MSBuild.MSBuildProjectSystem
        Unable to resolve assembly '/Users/df/git/test/Temp/bin/Debug/Assembly-CSharp-Editor-firstpass.dll'
[warn]: OmniSharp.MSBuild.MSBuildProjectSystem
        Unable to resolve assembly '/Users/df/git/test/Temp/bin/Debug/Assembly-CSharp-firstpass.dll'
[warn]: OmniSharp.MSBuild.MSBuildProjectSystem
        Unable to resolve assembly '/Users/df/git/test/Temp/bin/Debug/Assembly-CSharp.dll'
[info]: OmniSharp.Script.ScriptProjectSystem
        Detecting CSX files in '/Users/df/git/test'.
[info]: OmniSharp.Script.ScriptProjectSystem
        Could not find any CSX files
[info]: OmniSharp.Stdio.Host
        Invoking Workspace Options Provider: OmniSharp.Roslyn.CSharp.Services.CSharpWorkspaceOptionsProvider
[info]: OmniSharp.Stdio.Host
        Configuration finished.
[info]: OmniSharp.Stdio.Host
        Omnisharp server running using Stdio at location '/Users/df/git/test' on host 679.
```

**My previous mistake might be that I install dotnet after I opened up my project in VSCode**: when it perform an initial download of C# Debugger/OmniSharp server, it will fail at the last step, where it tries to run `dotnet --info`. I suspect this failure left some nasty state that requires me to do a clean install to recover.

ref: https://github.com/OmniSharp/omnisharp-vscode/blob/master/src/coreclr-debug/util.ts#L83-L118

**Comment by bitinn** [comment]
By the way, the "fire" icon is present: I am pretty sure it wasn't there between C# and feedback icon when I had the issue, I would have notice it otherwise.

<img width="1019" alt="screen shot 2018-02-13 at 14 56 44" src="https://user-images.githubusercontent.com/1484279/36137068-6da05e66-10ce-11e8-9925-9cbf3bea00eb.png">

**Comment by grofie** [comment]
Finally found the issue for my project: My solution files (xxx.sln, Assembly-CSharp.csproj, ...) were totally out of date. 

I deleted the solution files but could not get them back. Clicking Assets -> Open C# Project only opens VS Code but no new solution was created. So I had to install VS Community with the unity plugin in as well. Then I turned off VS Code and finally the solution files were generated again after pressing the Open C# Project button. Switched back to VS Code in the preferences and voilà, omnisharps works as expected.
